conductr_cli/conduct_main.py

```diff
@@ -270,19 +299,47 @@ def get_custom_settings(args):
 
 
 def run(_args=[]):
+    if not _args:
+        _args = sys.argv
+    # If we're being invoked via DC/OS then route our http
+    # calls via its extension to the requests library. In
+    # addition remove the dcos arg so that the conduct
+    # sub-commands are positioned correctly, along with their
+    # arguments.
+    if _args and Path(_args[0]).name == 'conduct.py':
+        _args = _args[1:]
+        dcos_mode = True
+    else:
+        dcos_mode = False
     # Parse arguments
-    parser = build_parser()
+    parser = build_parser(dcos_mode)
     argcomplete.autocomplete(parser)
-    args = parser.parse_args(_args) if _args else parser.parse_args()
+    args = parser.parse_args(_args[1:])
```

Comment:
We are removing here always the first argument. Also in non DCOS mode. Does that makes sense?